test/integration/java_purl_test.go

```diff
@@ -0,0 +1,219 @@
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+func TestJavaPURLs(t *testing.T) {
+	testImage := "anchore/test_images:java-56d52bc"
+	sbom := getCatalog(t, testImage)
+	found := make(map[string]string)
+	for _, p := range sbom.Artifacts.Packages.Sorted() {
+		if p.Type != pkg.JavaPkg && p.Type != pkg.JenkinsPluginPkg {
+			continue
+		}
+		key := fmt.Sprintf("%s@%s", p.Name, p.Version)
+		found[key] = p.PURL
+	}
+	for key, expectedPURL := range expectedPURLs {
+		purl := found[key]
+		assert.Equal(t, expectedPURL, purl, fmt.Sprintf("found wrong or missing PURL for %s want %s, got %s", key, expectedPURL, purl))
+	}
+	for key, foundPURL := range found {
+		expectedPURL := expectedPURLs[key]
+		assert.Equal(t, expectedPURL, foundPURL, fmt.Sprintf("found extra purl for %s want %s, got %s", key, expectedPURL, foundPURL))
+	}
+}
+
+func getCatalog(t *testing.T, image string) sbom.SBOM {
```

Comment:
TODO: this should be converted to use the normal test fixture mechanism.

Reply:
The Dockerfile in question is here: https://github.com/anchore/test-images/blob/main/containers/java/Dockerfile

I can probably just copy it into a test fixture.